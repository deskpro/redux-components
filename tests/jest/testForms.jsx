import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import {
  Form,
  Input,
  Checkbox,
  Datepicker,
  Radio,
  Select,
  Textarea,
  TagSet,
  FieldGroup
} from '../../src/index';

// https://github.com/facebook/react/issues/7740#issuecomment-247335106
const createNodeMock = () => ({
  addEventListener: () => {}
});
const mockStore = configureStore();

describe('>>> redux-form --- Snapshot', () => {
  it('+++capturing Snapshot of Input', () => {
    const renderedValue = renderer.create(<Provider store={mockStore()}>
      <Form name="test">
        <Input name="input1" />
      </Form>
                                          </Provider>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Checkbox', () => {
    const renderedValue = renderer.create(<Provider store={mockStore()}>
      <Form name="test">
        <Checkbox name="checkbox1" />
      </Form>
    </Provider>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Datepicker', () => {
    const renderedValue = renderer.create(
      <Provider store={mockStore()}>
        <Form name="test">
          <Datepicker name="date1" />
        </Form>
      </Provider>,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Radio', () => {
    const renderedValue = renderer.create(<Provider store={mockStore()}>
      <Form name="test">
        <Radio name="radio1" />
      </Form>
    </Provider>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Select', () => {
    const renderedValue = renderer.create(<Provider store={mockStore()}>
      <Form name="test">
        <Select
          name="select1"
          options={[
            { label: 'United Kingdom', value: 'uk' },
            { label: 'United States', value: 'us' }
          ]}
        />
      </Form>
    </Provider>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of Textarea', () => {
    const renderedValue = renderer.create(<Provider store={mockStore()}>
      <Form name="test">
        <Textarea name="textarea1" />
      </Form>
    </Provider>).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of TagSet', () => {
    const renderedValue = renderer.create(
      <Provider store={mockStore()}>
        <Form name="test">
          <TagSet name="tagset1" tags={['tag1', 'tag2']} />
        </Form>
      </Provider>,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  it('+++capturing Snapshot of FieldGroup', () => {
    const renderedValue = renderer.create(
      <Provider store={mockStore()}>
        <Form name="test">
          <FieldGroup label="test" meta={{ touched: false }}>
            <Input name="input1" />
          </FieldGroup>
        </Form>
      </Provider>,
      { createNodeMock }
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
