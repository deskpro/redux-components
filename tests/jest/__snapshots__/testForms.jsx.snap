// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Checkbox 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-input--checkbox"
      style={Object {}}
    >
      <input
        checked={false}
        disabled={false}
        id="id2"
        name="checkbox1"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="checkbox"
        value=""
      />
      <label
        className="dp-input--checkbox__checkbox"
        htmlFor="id2"
      />
      <label
        className="dp-input__label"
        htmlFor="id2"
      >
        
      </label>
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Datepicker 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-datepicker"
      style={Object {}}
    >
      <div
        className="dp-input dp-input--with-icon--right"
      >
        <i
          className="fa fa-calendar dp-icon dp-icon--s"
          onClick={[Function]}
        />
        <input
          id="id3"
          name="date1"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="DD/MM/YYYY"
          value=""
        />
      </div>
      <div />
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of FieldGroup 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    <label
      className="dp-input__label"
      htmlFor=""
    >
      test
      
    </label>
    
    <div
      className="dp-form-group"
    >
      
      
      <div
        className="dp-input"
      >
        <input
          id="id6"
          name="input1"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          value=""
        />
      </div>
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Input 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-input"
    >
      <input
        id="id1"
        name="input1"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        value=""
      />
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Radio 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-input--radio"
      style={Object {}}
    >
      <label
        htmlFor="id4"
      >
        <input
          checked={false}
          disabled={false}
          id="id4"
          name="radio1"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          type="radio"
          value=""
        />
        <i />
        <label
          className="dp-input__label"
          htmlFor="id4"
        >
          
          
        </label>
      </label>
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Select 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-select"
    >
      <div
        className="Select dp-select is-clearable is-searchable Select--single"
        style={undefined}
      >
        <div
          className="Select-control"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={undefined}
        >
          <span
            className="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              className="Select-placeholder"
            >
              Please select
            </div>
            <div
              className="Select-input"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-activedescendant="react-select-2--value"
                aria-describedby={undefined}
                aria-expanded="false"
                aria-haspopup="false"
                aria-label={undefined}
                aria-labelledby={undefined}
                aria-owns=""
                className={undefined}
                id={undefined}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                role="combobox"
                style={
                  Object {
                    "boxSizing": "content-box",
                    "width": "5px",
                  }
                }
                tabIndex={undefined}
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </span>
          <span
            className="Select-arrow-zone"
            onMouseDown={[Function]}
          >
            <span
              className="Select-arrow"
              onMouseDown={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of TagSet 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-tag-set"
    >
      <span
        className="dp-tag editable"
      >
        tag1
        <span
          onClick={[Function]}
        >
          <i
            className="fa fa-close dp-icon dp-icon--s dp-tag__close"
          />
        </span>
      </span>
      <span
        className="dp-tag editable"
      >
        tag2
        <span
          onClick={[Function]}
        >
          <i
            className="fa fa-close dp-icon dp-icon--s dp-tag__close"
          />
        </span>
      </span>
      <div
        className="dp-search-button"
        onChange={[Function]}
        onKeyDown={[Function]}
        style={undefined}
      >
        <button
          className="dp-button dp-button--l dp-button--primary dp-button--shape-default dp-button--square"
          disabled={false}
          onClick={[Function]}
        >
          <i
            className="fa fa-plus dp-icon dp-icon--s"
          />
        </button>
        <div />
      </div>
    </div>
  </div>
</form>
`;

exports[`>>> redux-form --- Snapshot +++capturing Snapshot of Textarea 1`] = `
<form
  className="dp-form"
  onSubmit={[Function]}
>
  <div
    className="dp-form-group"
  >
    
    
    <div
      className="dp-textarea"
      onBlur={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
    >
      <textarea
        name="textarea1"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
</form>
`;
